extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.Optimizely.js

```diff
@@ -0,0 +1,29 @@
+/* global toolbarABTestVariantNumber:false */
+/*!
+ * Utility functions for running Optimizely A/B tests with VisualEditor
+ */
+
+/**
+ * @returns {number} The A/B test variant number for the Toolbar A/B test
+ */
+ve.init.wikia.getToolbarABTestVariantNumber = function () {
+	// toolbarABTestVariantNumber is set for variant #1 for all Optimizely toolbar experiments
+	if ( typeof ( toolbarABTestVariantNumber ) === 'undefined' ) {
+		return 0;
+	}
+
+	return toolbarABTestVariantNumber;
+};
+
+/**
+ * Hook for activating experiment on VE load
+ */
+mw.hook( 've.activate' ).add( function () {
+	// Production tests are split by language. This is done because each language has a separate engagement percentage
+	var optimizelyIds = window.wgDevelEnvironment ?
+		[ '4721410313' ] : [ '4701112678', '5003080344', '5003533755', '5013830116' ];
+
+	optimizelyIds.forEach( function (element) {
+		window.optimizely.push( [ 'activate', element ] );
```

Comment:
So for all languages all experiments will be activated? And GA + Optimizely integration is somehow smart enough to figure out which experiment is should report in the custom values? (That's what I'm assuming it's going on because I don't see you reporting coversion within Optimizely - so it must be within GA)